### PR TITLE
Add date picker 

### PR DESCRIPTION
**Pull request by karthik-sista**
📘 [`CL-166`](https://osi-cwds.atlassian.net/browse/CL-166)

### CODE REVIEW

File: apps/www/package.json [hunk 1]
```diff
@@ -15,10 +15,12 @@
     "lodash.pick": "^4.4.0",
     "lodash.set": "^4.3.2",
     "react": "^16.4",
+    "react-datepicker": "^2.3.0",
```
Comment (mlipardo) [inline apps/www/package.json]: Should this be in forms libs?

File: apps/www/src/modules/components/articles/Forms/examples/DatePickerExample.jsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
+import React from 'react'
+import { Button, Card, CardHeader, CardTitle, CardBody } from '@cwds/components'
+import '@cwds/core/scss/styles.scss'
+import DS from '@cwds/core'
+import { Icon } from '@cwds/icons'
+import { Util } from '@cwds/reactstrap'
+import './DatePickerExampleStyles.scss'
+import DatePicker from 'react-datepicker'
+import 'react-datepicker/dist/react-datepicker.css'
```
Comment (mlipardo) [inline apps/www/src/modules/components/articles/Forms/examples/DatePickerExample.jsx]: this should import to `@cwds/components`

File: apps/www/package.json (hunk 1 above)
Comment (eckdanny-osi) [reply apps/www/package.json]: +1

File: apps/www/package.json [hunk 3]
```diff
@@ -15,10 +15,12 @@
...
     "react-dom": "^16.4",
     "react-router-dom": "^4.3.1",
     "react-scripts": "2.1.8",
     "react-testing-library": "^5.3.1",
+    "react-text-mask": "^5.4.3",
```
Comment (eckdanny-osi) [inline apps/www/package.json]: this dep should not be present in `www`. This dep should also be covered by the `InputMask` topic PR (not this PR)

File: apps/www/src/modules/components/articles/Forms/examples/DatePickerExample.jsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
+import React from 'react'
+import '@cwds/core/scss/styles.scss'
+import DS from '@cwds/core'
+import { Icon } from '@cwds/icons'
+import { Util } from '@cwds/reactstrap'
+import './DatePickerExampleStyles.scss'
+import DatePicker from 'react-datepicker'
```
Comment (eckdanny-osi) [inline apps/www/src/modules/components/articles/Forms/examples/DatePickerExample.jsx]: this should not be here. The `www` should only be importing `@cares/*` pkgs

File: apps/www/src/modules/components/articles/Forms/examples/DatePickerExample.jsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+import 'react-datepicker/dist/react-datepicker.css'
+import MaskedTextInput from 'react-text-mask'
```
Comment (eckdanny-osi) [inline apps/www/src/modules/components/articles/Forms/examples/DatePickerExample.jsx]: this should not be here. The `www` should only be importing `@cares/*` pkgs

File: apps/www/src/modules/components/articles/Forms/examples/DatePickerExample.jsx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+      <div className="row">
+        <div className="column">
+          <DatePicker
+            placeholderText="mm/dd/yyyy"
+            selected={this.state.startDate}
+            onChange={this.handleChange}
+            customInput={
+              <MaskedTextInput
+                type="text"
+                mask={[
```
Comment (eckdanny-osi) [inline apps/www/src/modules/components/articles/Forms/examples/DatePickerExample.jsx]: we can start a collection of _canned masks_ e.g.; in `@cwds/forms`. We don't want our consumers to craft their own masks... we do want them to use like

```jsx
import { MASKS } from '@cwds/forms'
....
<MaskedInput mask={MASKS.DATE_MASK} {...props} />
```

File: libs/forms/src/DatePicker.tsx [hunk 7]
```diff
@@ -0,0 +1,119 @@
+import React from 'react'
+import '@cwds/core/scss/styles.scss'
+import DS from '@cwds/core'
+import { Icon } from '@cwds/icons'
+import { Util } from '@cwds/reactstrap'
+import { IListType } from './types'
+
+import './DatePickerstyles.module.scss'
+import DatePicker from 'react-datepicker'
+import 'react-datepicker/dist/react-datepicker.css'
```
Comment (eckdanny-osi) [inline libs/forms/src/DatePicker.tsx]: let's manage style application consistently... if we're going to to piggyback on `react-datepicker`'s css, lets do it in the `*.module.scss` file for this component.

File: apps/www/src/modules/components/articles/Forms/examples/DatePickerExample.jsx [hunk 8]
```diff
@@ -0,0 +1,106 @@
+import React from 'react'
+import '@cwds/core/scss/styles.scss'
+import DS from '@cwds/core'
+import { Icon } from '@cwds/icons'
+import { Util } from '@cwds/reactstrap'
+import './DatePickerExampleStyles.scss'
```
Comment (eckdanny-osi) [inline apps/www/src/modules/components/articles/Forms/examples/DatePickerExample.jsx]: why would an Example need styles? We should rethink this

**Pull request merged**